sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/SqlScriptingParserSuite.scala

```diff
@@ -294,7 +294,21 @@ class SqlScriptingParserSuite extends SparkFunSuite with SQLHelper {
         parameters = Map("varName" -> "`testVariable`", "lineNumber" -> "4"))
   }
 
-  // TODO Add test for INVALID_VARIABLE_DECLARATION.NOT_ALLOWED_IN_SCOPE exception
+  test("declare in wrong scope") {
+    val sqlScriptText =
+      """
+        |BEGIN
+        | IF 1=1 THEN
+        |   DECLARE testVariable INTEGER;
+        | END IF;
+        |END""".stripMargin
+    checkError(
+      exception = intercept[SqlScriptingException] {
+        parseScript(sqlScriptText)
+      },
+      errorClass = "INVALID_VARIABLE_DECLARATION.NOT_ALLOWED_IN_SCOPE",
+      parameters = Map("varName" -> "`testVariable`", "lineNumber" -> "4"))
```

Comment:
Do you know why there is no query context here?

Reply:
as far as my understanding goes here, query context is the context of the expression/query plan, whereas SQL Scripting interpreter uses a new/separate node hierarchy (for both logical and execution nodes). for this purpose, we have created a new `SqlScriptingException` class that relies on `Origin` to create proper error messaging, i.e. exceptions in the "{LINE} [EXCEPTION_TYPE] exception message" format.